cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/DatasetAdminService.java

```diff
@@ -234,6 +234,8 @@ public DatasetAdmin call() throws Exception {
         });
       } catch (Exception e) {
         Throwables.propagateIfInstanceOf(e, IOException.class);
+        Throwables.propagateIfInstanceOf(e, DatasetManagementException.class);
```

Comment:
I saw that the enclosing method declared these 3 exceptions. Maybe the ds management exception was thrown below, I didn't check at the time.